.conan/test_package/CMakeLists.txt

```diff
@@ -4,9 +4,10 @@ cmake_minimum_required(VERSION 3.2.0 FATAL_ERROR)
 set(CMAKE_VERBOSE_MAKEFILE TRUE)
 
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup()
+conan_basic_setup(TARGETS)
 
 find_package(PEGTL REQUIRED CONFIG)
 
 add_executable(${PROJECT_NAME} test_package.cpp)
-target_link_libraries(${PROJECT_NAME} taocpp::pegtl)
+target_link_libraries(${PROJECT_NAME} CONAN_PKG::pegtl)
+set_target_properties(${PROJECT_NAME} PROPERTIES CXX_STANDARD 17)
```

Comment:
Why would setting the C++ standard be necessary here? The PEGTL is a header-only library that requires C++17, so C++17 should be pulled in from the previous line already, or am I misunderstanding how CMake works?


```suggestion

```

Reply:
I think you are right, lemme review the cmake recipe. If add some public property and export your target, so any project will be able to receive all configuration needed to be build. But if your project doesn't export anything and need some extra feature, like c++17, so your consumers will need to update their cmake files too.

Reply:
Done!